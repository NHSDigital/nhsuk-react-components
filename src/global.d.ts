declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
