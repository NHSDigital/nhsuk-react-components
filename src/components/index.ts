export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
