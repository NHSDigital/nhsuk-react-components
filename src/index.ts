export * from './components';
export * from './patterns';
