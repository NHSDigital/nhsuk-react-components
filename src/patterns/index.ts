export * from './nav-a-z/index.js';
export * from './review-date/index.js';
