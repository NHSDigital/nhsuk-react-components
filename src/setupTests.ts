import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
