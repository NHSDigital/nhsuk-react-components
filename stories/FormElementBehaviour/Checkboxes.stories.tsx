import React, { useEffect, SyntheticEvent } from 'react';
import { Checkboxes } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Checkboxes> = {
  title: 'FormBehaviour/Checkboxes',
  component: Checkboxes,
};
export default meta;
type Story = StoryObj<typeof Checkboxes>;

type CheckboxState = {
  box1: { name?: string; id?: string };
  box2: { name?: string; id?: string };
  box3: { name?: string; id?: string };
};

export const NoIDSupplied: Story = {
  render: function NoIDSuppliedRender() {
    const checkbox1Ref = React.useRef<HTMLInputElement>(null);
    const checkbox2Ref = React.useRef<HTMLInputElement>(null);
    const checkbox3Ref = React.useRef<HTMLInputElement>(null);

    const [checkboxState, setCheckboxState] = React.useState<CheckboxState>({
      box1: {
        name: '',
        id: '',
      },
      box2: {
        name: '',
        id: '',
      },
      box3: {
        name: '',
        id: '',
      },
    });

    useEffect(() => {
      setCheckboxState({
        box1: {
          name: checkbox1Ref.current?.name,
          id: checkbox1Ref.current?.id,
        },
        box2: {
          name: checkbox2Ref.current?.name,
          id: checkbox2Ref.current?.id,
        },
        box3: {
          name: checkbox3Ref.current?.name,
          id: checkbox3Ref.current?.id,
        },
      });
    }, [checkbox1Ref.current, checkbox2Ref.current, checkbox3Ref.current]);

    return (
      <div style={{ padding: 20 }}>
        <h2>Scenario: No ID Supplied</h2>
        <h5>Expected Behaviour</h5>
        <ul className="nhsuk-hint">
          <li>Boxes are selectable via their labels</li>
          <li>The Checkboxes are assigned IDs</li>
        </ul>
        <h5>Results</h5>
        <ul className="nhsuk-hint">
          <li>
            Box 1 ID:
            {checkboxState.box1.id}
          </li>
          <li>
            Box 2 ID:
            {checkboxState.box2.id}
          </li>
          <li>
            Box 3 ID:
            {checkboxState.box3.id}
          </li>
          <li>
            Box 1 Name:
            {checkboxState.box1.name}
          </li>
          <li>
            Box 2 Name:
            {checkboxState.box2.name}
          </li>
          <li>
            Box 3 Name:
            {checkboxState.box3.name}
          </li>
        </ul>
        <h5>Component</h5>
        <Checkboxes>
          <Checkboxes.Box inputRef={checkbox1Ref}>Box 1</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox2Ref}>Box 2</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox3Ref}>Box 3</Checkboxes.Box>
        </Checkboxes>
      </div>
    );
  },
};

export const NameSupplied: Story = {
  render: function NameSuppliedRender() {
    const checkbox1Ref = React.useRef<HTMLInputElement>(null);
    const checkbox2Ref = React.useRef<HTMLInputElement>(null);
    const checkbox3Ref = React.useRef<HTMLInputElement>(null);

    const [checkboxState, setCheckboxState] = React.useState<CheckboxState>({
      box1: {
        name: '',
        id: '',
      },
      box2: {
        name: '',
        id: '',
      },
      box3: {
        name: '',
        id: '',
      },
    });

    useEffect(() => {
      setCheckboxState({
        box1: {
          name: checkbox1Ref.current?.name,
          id: checkbox1Ref.current?.id,
        },
        box2: {
          name: checkbox2Ref.current?.name,
          id: checkbox2Ref.current?.id,
        },
        box3: {
          name: checkbox3Ref.current?.name,
          id: checkbox3Ref.current?.id,
        },
      });
    }, [checkbox1Ref.current, checkbox2Ref.current, checkbox3Ref.current]);

    return (
      <div style={{ padding: 20 }}>
        <h2>Scenario: Name Supplied</h2>
        <h5>Expected Behaviour</h5>
        <ul className="nhsuk-hint">
          <li>Boxes are selectable via their labels</li>
          <li>The Checkboxes are assigned IDs according to the Checkboxes Name</li>
        </ul>
        <h5>Results</h5>
        <ul className="nhsuk-hint">
          <li>
            Box 1 ID:
            {checkboxState.box1.id}
          </li>
          <li>
            Box 2 ID:
            {checkboxState.box2.id}
          </li>
          <li>
            Box 3 ID:
            {checkboxState.box3.id}
          </li>
          <li>
            Box 1 Name:
            {checkboxState.box1.name}
          </li>
          <li>
            Box 2 Name:
            {checkboxState.box2.name}
          </li>
          <li>
            Box 3 Name:
            {checkboxState.box3.name}
          </li>
        </ul>
        <h5>Component</h5>
        <Checkboxes name="name-supplied">
          <Checkboxes.Box inputRef={checkbox1Ref}>Box 1</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox2Ref}>Box 2</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox3Ref}>Box 3</Checkboxes.Box>
        </Checkboxes>
      </div>
    );
  },
};

export const IDPrefixSupplied: Story = {
  render: function IDPrefixSuppliedRender() {
    const checkbox1Ref = React.useRef<HTMLInputElement>(null);
    const checkbox2Ref = React.useRef<HTMLInputElement>(null);
    const checkbox3Ref = React.useRef<HTMLInputElement>(null);

    const [checkboxState, setCheckboxState] = React.useState<CheckboxState>({
      box1: {
        name: '',
        id: '',
      },
      box2: {
        name: '',
        id: '',
      },
      box3: {
        name: '',
        id: '',
      },
    });

    useEffect(() => {
      setCheckboxState({
        box1: {
          name: checkbox1Ref.current?.name,
          id: checkbox1Ref.current?.id,
        },
        box2: {
          name: checkbox2Ref.current?.name,
          id: checkbox2Ref.current?.id,
        },
        box3: {
          name: checkbox3Ref.current?.name,
          id: checkbox3Ref.current?.id,
        },
      });
    }, [checkbox1Ref.current, checkbox2Ref.current, checkbox3Ref.current]);

    return (
      <div style={{ padding: 20 }}>
        <h2>Scenario: ID Prefix Supplied</h2>
        <h5>Expected Behaviour</h5>
        <ul className="nhsuk-hint">
          <li>Boxes are selectable via their labels</li>
          <li>The Checkboxes are assigned IDs according to the ID prefix</li>
          <li>The Checkboxes are assigned randomly generated names</li>
        </ul>
        <h5>Results</h5>
        <ul className="nhsuk-hint">
          <li>
            Box 1 ID:
            {checkboxState.box1.id}
          </li>
          <li>
            Box 2 ID:
            {checkboxState.box2.id}
          </li>
          <li>
            Box 3 ID:
            {checkboxState.box3.id}
          </li>
          <li>
            Box 1 Name:
            {checkboxState.box1.name}
          </li>
          <li>
            Box 2 Name:
            {checkboxState.box2.name}
          </li>
          <li>
            Box 3 Name:
            {checkboxState.box3.name}
          </li>
        </ul>
        <h5>Component</h5>
        <Checkboxes idPrefix="idprefix">
          <Checkboxes.Box inputRef={checkbox1Ref}>Box 1</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox2Ref}>Box 2</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox3Ref}>Box 3</Checkboxes.Box>
        </Checkboxes>
      </div>
    );
  },
};

export const IDPrefixAndNameSupplied: Story = {
  render: function IDPrefixAndNameSuppliedRender() {
    const checkbox1Ref = React.useRef<HTMLInputElement>(null);
    const checkbox2Ref = React.useRef<HTMLInputElement>(null);
    const checkbox3Ref = React.useRef<HTMLInputElement>(null);

    const [checkboxState, setCheckboxState] = React.useState<CheckboxState>({
      box1: {
        name: '',
        id: '',
      },
      box2: {
        name: '',
        id: '',
      },
      box3: {
        name: '',
        id: '',
      },
    });

    useEffect(() => {
      setCheckboxState({
        box1: {
          name: checkbox1Ref.current?.name,
          id: checkbox1Ref.current?.id,
        },
        box2: {
          name: checkbox2Ref.current?.name,
          id: checkbox2Ref.current?.id,
        },
        box3: {
          name: checkbox3Ref.current?.name,
          id: checkbox3Ref.current?.id,
        },
      });
    }, [checkbox1Ref.current, checkbox2Ref.current, checkbox3Ref.current]);

    return (
      <div style={{ padding: 20 }}>
        <h2>Scenario: ID Prefix and Name Supplied</h2>
        <h5>Expected Behaviour</h5>
        <ul className="nhsuk-hint">
          <li>Boxes are selectable via their labels</li>
          <li>The Checkboxes are assigned IDs according to the ID prefix</li>
          <li>The Checkboxes have the same name as the Checkboxes component</li>
        </ul>
        <h5>Results</h5>
        <ul className="nhsuk-hint">
          <li>
            Box 1 ID:
            {checkboxState.box1.id}
          </li>
          <li>
            Box 2 ID:
            {checkboxState.box2.id}
          </li>
          <li>
            Box 3 ID:
            {checkboxState.box3.id}
          </li>
          <li>
            Box 1 Name:
            {checkboxState.box1.name}
          </li>
          <li>
            Box 2 Name:
            {checkboxState.box2.name}
          </li>
          <li>
            Box 3 Name:
            {checkboxState.box3.name}
          </li>
        </ul>
        <h5>Component</h5>
        <Checkboxes idPrefix="idprefix" name="testname">
          <Checkboxes.Box inputRef={checkbox1Ref}>Box 1</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox2Ref}>Box 2</Checkboxes.Box>
          <Checkboxes.Box inputRef={checkbox3Ref}>Box 3</Checkboxes.Box>
        </Checkboxes>
      </div>
    );
  },
};

export const OnChangeAndOnInputHandlers: Story = {
  render: function OnChangeAndOnInputHandlersRender() {
    const [changeEventLog, setChangeEventLog] = React.useState<Array<string>>([]);
    const [inputEventLog, setInputEventLog] = React.useState<Array<string>>([]);
    const [currentValue, setCurrentValue] = React.useState<Array<string>>([]);

    const handleChange = (e: SyntheticEvent<HTMLInputElement>) => {
      const target = e.target as HTMLInputElement;
      setChangeEventLog([
        ...changeEventLog,
        `[${target.id}] Value: ${target.value}, Checked: ${target.checked}`,
      ]);
      if (currentValue.includes(target.value)) {
        setCurrentValue(currentValue.filter((x) => x !== target.value));
      } else {
        setCurrentValue([...currentValue, target.value]);
      }
    };

    const handleInput = (e: SyntheticEvent<HTMLInputElement>) => {
      const target = e.target as HTMLInputElement;
      setInputEventLog([
        ...inputEventLog,
        `[${target.id}] Value: ${target.value}, Checked: ${target.checked}`,
      ]);
    };

    return (
      <div style={{ padding: 20 }}>
        <h2>Scenario: onChange and onInput handlers are bound without any other props</h2>
        <h5>Expected Behaviour</h5>
        <ul className="nhsuk-hint">
          <li>OnChange Handlers are fired using the generated IDs and Names</li>
          <li>The value is passed through</li>
        </ul>
        <h5>Component</h5>
        <Checkboxes onChange={handleChange} onInput={handleInput}>
          <Checkboxes.Box value="1">Box 1</Checkboxes.Box>
          <Checkboxes.Box value="2">Box 2</Checkboxes.Box>
          <Checkboxes.Box value="3">Box 3</Checkboxes.Box>
        </Checkboxes>
        <br />
        <h5>Current Value</h5>
        <pre>{JSON.stringify(currentValue, null, 2)}</pre>
        <h5>Change Events</h5>
        <ul className="nhsuk-hint">
          {changeEventLog.map((event, index) => (
            <li key={index}>{event}</li>
          ))}
        </ul>
        <h5>Input Events</h5>
        <ul className="nhsuk-hint">
          {inputEventLog.map((event, index) => (
            <li key={index}>{event}</li>
          ))}
        </ul>
      </div>
    );
  },
};
