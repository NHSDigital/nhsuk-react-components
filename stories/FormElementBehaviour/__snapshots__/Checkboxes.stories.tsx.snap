// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBehaviour/Checkboxes IDPrefixAndNameSupplied smoke-test 1`] = `
<div style="padding: 20px;">
  <h2>
    Scenario: ID Prefix and Name Supplied
  </h2>
  <h5>
    Expected Behaviour
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Boxes are selectable via their labels
    </li>
    <li>
      The Checkboxes are assigned IDs according to the ID prefix
    </li>
    <li>
      The Checkboxes have the same name as the Checkboxes component
    </li>
  </ul>
  <h5>
    Results
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Box 1 ID:idprefix-1
    </li>
    <li>
      Box 2 ID:idprefix-2
    </li>
    <li>
      Box 3 ID:idprefix-3
    </li>
    <li>
      Box 1 Name:testname
    </li>
    <li>
      Box 2 Name:testname
    </li>
    <li>
      Box 3 Name:testname
    </li>
  </ul>
  <h5>
    Component
  </h5>
  <div class="nhsuk-form-group">
    <div class="nhsuk-checkboxes"
         id="checkboxes_qr2rn"
    >
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="testname"
               id="idprefix-1"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-1--label"
               for="idprefix-1"
        >
          Box 1
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="testname"
               id="idprefix-2"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-2--label"
               for="idprefix-2"
        >
          Box 2
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="testname"
               id="idprefix-3"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-3--label"
               for="idprefix-3"
        >
          Box 3
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormBehaviour/Checkboxes IDPrefixSupplied smoke-test 1`] = `
<div style="padding: 20px;">
  <h2>
    Scenario: ID Prefix Supplied
  </h2>
  <h5>
    Expected Behaviour
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Boxes are selectable via their labels
    </li>
    <li>
      The Checkboxes are assigned IDs according to the ID prefix
    </li>
    <li>
      The Checkboxes are assigned randomly generated names
    </li>
  </ul>
  <h5>
    Results
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Box 1 ID:idprefix-1
    </li>
    <li>
      Box 2 ID:idprefix-2
    </li>
    <li>
      Box 3 ID:idprefix-3
    </li>
    <li>
      Box 1 Name:checkboxes_uilvo
    </li>
    <li>
      Box 2 Name:checkboxes_uilvo
    </li>
    <li>
      Box 3 Name:checkboxes_uilvo
    </li>
  </ul>
  <h5>
    Component
  </h5>
  <div class="nhsuk-form-group">
    <div class="nhsuk-checkboxes"
         id="checkboxes_uilvo"
    >
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_uilvo"
               id="idprefix-1"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-1--label"
               for="idprefix-1"
        >
          Box 1
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_uilvo"
               id="idprefix-2"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-2--label"
               for="idprefix-2"
        >
          Box 2
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_uilvo"
               id="idprefix-3"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="idprefix-3--label"
               for="idprefix-3"
        >
          Box 3
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormBehaviour/Checkboxes NameSupplied smoke-test 1`] = `
<div style="padding: 20px;">
  <h2>
    Scenario: Name Supplied
  </h2>
  <h5>
    Expected Behaviour
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Boxes are selectable via their labels
    </li>
    <li>
      The Checkboxes are assigned IDs according to the Checkboxes Name
    </li>
  </ul>
  <h5>
    Results
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Box 1 ID:checkboxes_nuwkl-1
    </li>
    <li>
      Box 2 ID:checkboxes_nuwkl-2
    </li>
    <li>
      Box 3 ID:checkboxes_nuwkl-3
    </li>
    <li>
      Box 1 Name:name-supplied
    </li>
    <li>
      Box 2 Name:name-supplied
    </li>
    <li>
      Box 3 Name:name-supplied
    </li>
  </ul>
  <h5>
    Component
  </h5>
  <div class="nhsuk-form-group">
    <div class="nhsuk-checkboxes"
         id="checkboxes_nuwkl"
    >
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="name-supplied"
               id="checkboxes_nuwkl-1"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_nuwkl-1--label"
               for="checkboxes_nuwkl-1"
        >
          Box 1
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="name-supplied"
               id="checkboxes_nuwkl-2"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_nuwkl-2--label"
               for="checkboxes_nuwkl-2"
        >
          Box 2
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="name-supplied"
               id="checkboxes_nuwkl-3"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_nuwkl-3--label"
               for="checkboxes_nuwkl-3"
        >
          Box 3
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormBehaviour/Checkboxes NoIDSupplied smoke-test 1`] = `
<div style="padding: 20px;">
  <h2>
    Scenario: No ID Supplied
  </h2>
  <h5>
    Expected Behaviour
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Boxes are selectable via their labels
    </li>
    <li>
      The Checkboxes are assigned IDs
    </li>
  </ul>
  <h5>
    Results
  </h5>
  <ul class="nhsuk-hint">
    <li>
      Box 1 ID:checkboxes_f1vbb-1
    </li>
    <li>
      Box 2 ID:checkboxes_f1vbb-2
    </li>
    <li>
      Box 3 ID:checkboxes_f1vbb-3
    </li>
    <li>
      Box 1 Name:checkboxes_f1vbb
    </li>
    <li>
      Box 2 Name:checkboxes_f1vbb
    </li>
    <li>
      Box 3 Name:checkboxes_f1vbb
    </li>
  </ul>
  <h5>
    Component
  </h5>
  <div class="nhsuk-form-group">
    <div class="nhsuk-checkboxes"
         id="checkboxes_f1vbb"
    >
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_f1vbb"
               id="checkboxes_f1vbb-1"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_f1vbb-1--label"
               for="checkboxes_f1vbb-1"
        >
          Box 1
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_f1vbb"
               id="checkboxes_f1vbb-2"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_f1vbb-2--label"
               for="checkboxes_f1vbb-2"
        >
          Box 2
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_f1vbb"
               id="checkboxes_f1vbb-3"
               type="checkbox"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_f1vbb-3--label"
               for="checkboxes_f1vbb-3"
        >
          Box 3
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormBehaviour/Checkboxes OnChangeAndOnInputHandlers smoke-test 1`] = `
<div style="padding: 20px;">
  <h2>
    Scenario: onChange and onInput handlers are bound without any other props
  </h2>
  <h5>
    Expected Behaviour
  </h5>
  <ul class="nhsuk-hint">
    <li>
      OnChange Handlers are fired using the generated IDs and Names
    </li>
    <li>
      The value is passed through
    </li>
  </ul>
  <h5>
    Component
  </h5>
  <div class="nhsuk-form-group">
    <div class="nhsuk-checkboxes"
         id="checkboxes_7h4nb"
    >
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_7h4nb"
               id="checkboxes_7h4nb-1"
               type="checkbox"
               value="1"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_7h4nb-1--label"
               for="checkboxes_7h4nb-1"
        >
          Box 1
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_7h4nb"
               id="checkboxes_7h4nb-2"
               type="checkbox"
               value="2"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_7h4nb-2--label"
               for="checkboxes_7h4nb-2"
        >
          Box 2
        </label>
      </div>
      <div class="nhsuk-checkboxes__item">
        <input class="nhsuk-checkboxes__input"
               name="checkboxes_7h4nb"
               id="checkboxes_7h4nb-3"
               type="checkbox"
               value="3"
        >
        <label class="nhsuk-label nhsuk-checkboxes__label"
               id="checkboxes_7h4nb-3--label"
               for="checkboxes_7h4nb-3"
        >
          Box 3
        </label>
      </div>
    </div>
  </div>
  <br>
  <h5>
    Current Value
  </h5>
  <pre>
    []
  </pre>
  <h5>
    Change Events
  </h5>
  <ul class="nhsuk-hint">
  </ul>
  <h5>
    Input Events
  </h5>
  <ul class="nhsuk-hint">
  </ul>
</div>
`;
